### PR TITLE
Added BaseColorService similar to AccentColorService

### PR DESCRIPTION
**Pull request by bigworld12**
### Description of Change ###
`Light` base theme was hard-coded into `ThemeService`, so i added a `BaseColorService` to select base themes

### API Changes ###
- Added `BaseColorService` and integrated it into `ThemeService`

### Platforms Affected ### 
- WPF

### Behavioral Changes ###
None

### PR Checklist ###

- [x] I have included examples or tests
- [ ] I have updated the change log
- [ ] I am listed in the CONTRIBUTORS file
- [ ] Rebased on top of the target branch at time of PR
- [ ] Changes adhere to coding standard

### CODE REVIEW

File: src/Orchestra.Examples.Shared/Views/ControlsView.xaml [hunk 1]
```diff
@@ -101,8 +120,8 @@
 
       <GroupBox Header="Orc controls">
         <WrapPanel>
-            <orccontrols:BusyIndicator ToolTip="Orchestra: BusyIndicator" Width="100" Height="12" />
-            <orccontrols:FluidProgressBar ToolTip="Orchestra: FluidProgressBar" Width="100" Height="12" />
+          <orccontrols:BusyIndicator ToolTip="Orchestra: BusyIndicator" Width="100" Height="12" />
```
Comment (bigworld12) [inline src/Orchestra.Examples.Shared/Views/ControlsView.xaml]: god damn i hate indentation

File: src/Orchestra.Core/Services/BaseColorService.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    public class BaseColorService : IBaseColorService
+    {
+        public event EventHandler<EventArgs> BaseColorChanged;
+        
+        private string _baseColor = null;
+        public string GetBaseColor() => _baseColor ?? (_baseColor = GetAvailableBaseColors()[0]);
+        public bool SetBaseColor(string color)
+        {
+            if (_baseColor == color || !GetAvailableBaseColors().Contains(color))
```
Comment (GeertvanHorrik) [inline src/Orchestra.Core/Services/BaseColorService.cs]: `_baseColor.EqualsIgnoreCase(color)`

and please use { and } to wrap the if logic.

File: src/Orchestra.Examples.Shared/ViewModels/ControlsViewModel.cs [hunk 3]
```diff
@@ -7,32 +7,45 @@
...
         public Color SelectedAccentColor { get; set; }
+        public string SelectedBaseColor { get; set; }
         #endregion
 
         #region Methods
         private void OnSelectedAccentColorChanged()
         {
             _accentColorService.SetAccentColor(SelectedAccentColor);
         }
+        private void OnSelectedBaseColorChanged()
```
Comment (GeertvanHorrik) [inline src/Orchestra.Examples.Shared/ViewModels/ControlsViewModel.cs]: keep white lines between methods.

File: src/Orchestra.Shell.MahApps/Services/MahAppsBaseColorService.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Orchestra.Services
+{
+    using System.Collections.Generic;
+    public class MahAppsBaseColorService : BaseColorService
+    {
+        public override IReadOnlyList<string> GetAvailableBaseColors()
+        {
+            return new List<string>() { OrchestraEnvironment.DefaultBaseColor, "Dark" }.AsReadOnly();
```
Comment (GeertvanHorrik) [inline src/Orchestra.Shell.MahApps/Services/MahAppsBaseColorService.cs]: Please use "Light", "Dark". For example, what if we change the defaultbase color to dark isntead of light?

File: src/Orchestra.Core/Services/BaseColorService.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+                return false;
+            _baseColor = color;
+            BaseColorChanged?.Invoke(this, EventArgs.Empty);
+            return true;
+        }
+       
+        public virtual IReadOnlyList<string> GetAvailableBaseColors()
+        {
+            return new List<string>() { OrchestraEnvironment.DefaultBaseColor }.AsReadOnly();
```
Comment (GeertvanHorrik) [inline src/Orchestra.Core/Services/BaseColorService.cs]: Please use "Light". For example, what if we change the defaultbase color to dark isntead of light?

File: src/Orchestra.Core/Services/BaseColorService.cs (hunk 2 above)
Comment (bigworld12) [reply src/Orchestra.Core/Services/BaseColorService.cs]: Not sure if the new changes match what you want, can you please check ?

**Comment by GeertvanHorrik** [comment]
Perfect, thanks a lot!

**Pull request merged**